javaslang/src/main/java/javaslang/collection/Iterator.java

```diff
@@ -1535,9 +1537,9 @@ default T get() {
 
     @Override
     default Iterator<Seq<T>> grouped(int size) {
```

Comment:
Scala's grouped() and sliding() functions return a `GroupedIterator<Seq<T>>`. That GroupedIterator has the ability of padding and skipping partial results (both not implemented in our version, yet).

It would be nice if we could provide the same functionality. Returning a `GroupedIterator<Seq<T>>` instead of `Iterator<Seq<T>>` will not be backward compatible. I will target that change for 3.0.0